src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.step.finscholar.firebase;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.WriteResult;
+import com.google.firebase.FirebaseException;
+import com.google.gson.Gson;
+import com.google.step.finscholar.data.ServletConstantValues;
+import java.util.logging.Logger;
+
+/** This class handles writing to and reading from a Cloud Firestore database. */
+public class FirebaseStorageManager {
+  /** This is used to convert Objects into JSON strings to send to the frontend. */
+  private static Gson gson = new Gson();
+
+  /** Formatters to use with String.format(). */
+  private static final String ADDED_NEW_DOC_FORMATTER = "%s%s%s%s";
```

Comment:
Instead of this a bunch of %s's, you can remove some of the constants you have. So it would be

```
"New document added to %s at %s."
```